example-app/App.tsx

```diff
@@ -40,14 +40,45 @@ const Header = () => {
 };
 
 const App = () => {
+  const [result, setResult] = React.useState<DeepLinkResult>(undefined);
+
+  React.useEffect(() => {
+    Linking.addEventListener('url', event => {
+      setResult(handleDeepLink(event.url));
+    });
+
+    async function getDeepLink() {
+      let url = await Linking.getInitialURL();
+      let res = handleDeepLink(url);
+      setResult(res);
+    }
+
+    getDeepLink();
+  }, []);
+
+  const handleDeepLink = (url: string | null): DeepLinkResult => {
+    if (!url) {
+      return;
+    }
+
+    let parsedQs = qs.parse(url.substring(url.indexOf('?') + 1));
```

Comment:
Use URL to parse instead of substring? It is not good practice to parse URLs yourself.

Reply:
I'll change it back to what it was before with url-parse